### ISSUE TITLE
Responsive Maps

### ISSUE THREAD

**Issue by MarcSkovMadsen** [developmentseed/lonboard#264]
When creating data apps as in #262 you most often want responsive maps that fill out the parent container and adjusts if its size changes.

I don't how this is possible with lonboard. I've looked through the documentation.

Please add. Thanks.

**Comment by kylebarron** [comment]
I agree this would be great, but I'm not really a web developer and don't know how to achieve it. We tried to do this in https://github.com/developmentseed/lonboard/pull/220 and https://github.com/developmentseed/lonboard/issues/167, but only tested in jupyterlab. I don't know a general solution

**Comment by batpad** [comment]
A new version of `lonboard` has been released with the improvements to getting the height to fill up correctly. This has fixed it for some cases I saw where the height would behave strangely. @MarcSkovMadsen are you able to check your case with the latest version to see if the height filling is improved? And if it's not, if you're able to share a bit more of what you're trying / a screenshot of the problem or so - thanks much!

**Comment by wx4stg** [comment]
I'm running into this issue, specifically when using panel, and it seems to be a change introduced by lonboard 0.10. I have the following example which is taken (almost, had to modify the URL) directly from the tutorial of panel+lonboard:

<details>
<summary> Code Example </summary>

```
import geopandas as gpd
from lonboard import Map, ScatterplotLayer
import panel as pn

pn.extension("ipywidgets")

colors = {
    "Red": [200, 0, 0],
    "Green": [0, 200, 0],
    "Blue": [0, 0, 200],
}

@pn.cache
def get_data():
    return gpd.read_file("https://naciscdn.org/naturalearth/110m/cultural/ne_110m_populated_places.zip")

gdf = get_data()
layer = ScatterplotLayer.from_geopandas(gdf, radius_min_pixels=2, get_fill_color="red")
cities_map = Map(layer)

# Fit to the available space
cities_map.layout.height = cities_map.layout.width = "100%"

color_input = pn.widgets.Select(
    name="Color", options=list(colors.keys()), description="The color of the points"
)

@pn.depends(value=color_input, watch=True)
def set_fill_color(value):
    cities_map.layers[0].get_fill_color = colors[value]

pn.Column(
    color_input,
    pn.pane.IPyWidget(cities_map, sizing_mode="stretch_both"),
).save('lon0.10.3.html')
```

</details>

On version 0.9.3: https://wx4stg.com/public/lon0.9.3.html
On version 0.10.3: https://wx4stg.com/public/lon0.10.3.html

The difference in the resulting HTML seems to come from something to do with this line
`<div id="map-1db471f8-47a4-417f-acd3-8951731a61a8" style="height: 100%;">` (in 0.9.3)
`<div id="map-bbe0dfc7-8208-45ca-87e2-ee6b5b19490c" class="flex" style="height: 24rem;">` (in 0.10.3)

in addition to the height change, the map object is contained in another `<div>` element. Interestingly, I tried inspect-element changing the "height: 24rem;" to "height: 100%;", but that just causes the map view to disappear completely.


_Obligatory: I am raising this the holidays, I finally have time to work on some previously-tabled side projects involving lonboard, but please PLEASE do not feel any need to respond anytime soon. I recognize my schedule (grad student) is not the same as anyone elses'. Thanks!_

**Comment by kylebarron** [comment]
I'm not deeply familiar with CSS and HTML styling. Potentially it's a regression caused by some CSS changes in @vgeorge 's work.

I think it would be easier if you opened a new issue and linked back to this one. Are you able to track down a specific PR that caused the changes?
